### DISCUSSION TITLE
Commit Limit 100MB

### DISCUSSION THREAD

**Discussion by ToptenDev** [community/community#49324] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Question

### Body

I am getting git commit size exceeds error for 100MB. I tried to research myself but not result yet.

**Comment by kuiux** [comment, marked as answer]
If you have checked all the files in your repository using git ls-files and found that none of them exceed the 100 MB limit, it's possible that the error is caused by a commit that is being pushed and contains a large file.

In this case, you can use the git rev-list command to list the commits in your local repository and their sizes. The git rev-list command shows the commit hash, the author, the date, and the commit message, as well as the size of the commit.

To do this, run the following command:

css
Copy code
git rev-list --objects --all | sort -k 2 > allfileshas.txt
This will create a file called allfileshas.txt that lists all the objects in your repository along with their size. You can then review this file to identify any large commits that may be causing the issue.

If you find a large commit, you can use the git filter-branch command to remove the large file from the commit history. You will need to force push your changes after running this command, so make sure to communicate this to any collaborators who may be working on the repository.

Note that if you do remove a large file from your repository history, anyone who has already cloned the repository will need to re-clone it to get the updated history.

**Comment by queenofcorgis** [comment]
Thanks for posting in the GitHub Community, @ToptenDev ! 

We're happy you're here. You are more likely to get a useful response if you are posting your question in the applicable category, the Issues category is solely related to conversations around the GitHub products Issues and Projects. I've gone ahead and moved it for you. Good luck!
